src/app/ui/timeline/timeline.cpp

```diff
@@ -4629,25 +4629,37 @@ void Timeline::onDrop(ui::DragEvent& e)
     auto surface = e.getImage();
 
     execute_from_ui_thread([=] {
-      std::string txmsg;
-      std::unique_ptr<docapi::DocProvider> docProvider = nullptr;
-      if (droppedImage) {
-        txmsg = "Drop Image";
-        doc::ImageRef image = nullptr;
-        convert_surface_to_image(surface.get(), 0, 0, surface->width(), surface->height(), image);
-        docProvider = std::make_unique<DocProviderFromImage>(image);
+      try {
+        std::string txmsg;
+        std::unique_ptr<docapi::DocProvider> docProvider = nullptr;
+        if (droppedImage) {
+          if (!surface)
+            throw std::runtime_error("Invalid format");
+
+          txmsg = "Drop Image";
+          doc::ImageRef image = nullptr;
+          convert_surface_to_image(surface.get(), 0, 0, surface->width(), surface->height(), image);
+          docProvider = std::make_unique<DocProviderFromImage>(image);
+        }
+        else {
+          txmsg = "Drop File";
+          docProvider = std::make_unique<DocProviderFromPaths>(m_document->context(), paths);
+        }
+
+        Tx tx(m_document, txmsg);
+        DocApi docApi(m_document, tx);
+        docApi
+          .dropDocumentsOnTimeline(m_document, frame, layerIndex, insert, droppedOn, *docProvider);
+        tx.commit();
+        m_document->notifyGeneralUpdate();
       }
-      else {
-        txmsg = "Drop File";
-        docProvider = std::make_unique<DocProviderFromPaths>(m_document->context(), paths);
+      catch (const std::exception& e) {
+        Console::showException(e);
+        Console::window()->invalidate();
+        Console::window()->flushRedraw();
```

Comment:
I think we shouldn't give access to the internal console window pointer (at least not for this) and just fix the `showException()` if it needs these function calls inside.

Reply:
Well, actually it is needed for any case where the showException is called from within the onDrop method. For instance, try to drag and drop a pdf file into the timeline. You will see that the console shows up, but it is empty until you click on it. On the other hand, if you drag and drop a pdf into Aseprite, it will show the console window properly.